packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/library/services/LibrarySearchService.java

```diff
@@ -61,6 +62,16 @@ public Optional<Library> findById(UUID id) {
         return this.libraryRepository.findById(id);
     }
 
+    @Override
+    public List<Library> findAllById(Iterable<UUID> ids) {
+        return this.libraryRepository.findAllById(ids);
+    }
+
+    @Override
+    public Optional<Library> findByDocumentId(UUID documentId) {
+        return this.libraryRepository.findByDocumentId(documentId);
+    }
```

Comment:
- The concept of document does not exists in the bounded context of the library, this highlight a lifecycle issue
- Even if it did, you cannot find the root of an aggregate using the identifier of one of its child element. In this case, the id of a document is not unique to a semantic data and thus to a library. Multiple libraries WILL have documents with the same id.

Reply:
This method doesn't exist anymore